### PR TITLE
Added new Checks

### PR DESCRIPTION
**Pull request by H31S3nb3rg-0x00**
Check for get permissions  ( has to be only read)
Check for post permissions ( has to be only read and write)
Check for put permissions ( has to be only write)

### CODE REVIEW

File: swagger/src/scan/checks.rs [hunk 1]
```diff
@@ -54,7 +54,10 @@ impl_passive_checks![
     (CheckStrAttrs,check_str_attrs,"STRING ATTRIBUTES","Checks for the definion of string type attributes - max_length, min_length, pattern"),
     (CheckArrAttrs,check_arr_attrs,"ARRAY ATTRIBUTES","Checks for the definion of array type attributes - max_items, min_items"),
     (CheckObjAttrs,check_obj_attrs,"OBJECT ATTRIBUTES","Checks for the definion of object type attributes - max_properties, properties"),
-    (CheckValidResponses,check_valid_responses,"VALID RESPONSES","Checks for valid responses codes")
+    (CheckValidResponses,check_valid_responses,"VALID RESPONSES","Checks for valid responses codes"),
+    (CheckGetPermissions, check_get_permissions, "GET PERMISSIONS", "Checks for correct permissions for get requests"),
```
Comment (GuyL99) [inline swagger/src/scan/checks.rs:58]: We may need to merge those 3 checks into a single check with 3 alerts

**Review by GuyL99 — CHANGES_REQUESTED** [review]
We'll revisit it later next week

File: swagger/src/scan/passive/additional_checks.rs [hunk 2]
```diff
@@ -23,4 +23,94 @@ impl<T: OAS + Serialize> PassiveSwaggerScan<T> {
         }
         alerts
     }
+    pub fn check_get_permissions(&self) -> Vec<Alert> {
+        let mut alerts: Vec<Alert> = vec![];
+        for (path, item) in &self.swagger.get_paths() {
+            for(m,op) in item.get_ops(){       
+                if m == Method::GET{
+                    match &op.security {
+                        Some(x) => {
```
[the remark this answers is not in the document]
Comment (H31S3nb3rg-0x00) [reply swagger/src/scan/passive/additional_checks.rs:32]: How do you suggest to name schemes?

**Comment by H31S3nb3rg-0x00** [comment]
I would happy to revisit my code together in order to apply changes

**Pull request merged**